Add a link to available jinja2 playbook filters

rsicart opened the pull request:
##### SUMMARY
Add a link to available jinja2 playbook filters.
Useful to find quilckly what kind of expression we can use in this option.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
plugin/doc_fragments/constructed

##### ADDITIONAL INFORMATION
I created a PR https://github.com/ansible-collections/google.cloud/pull/296 to write an example of use.

rsicart committed:
Add a link to available jinja2 playbook filters

bcoca commented:
why not tests also? this seems a bit superflous, I would point to Jinja docs in general or multiple links, also to all fields that allow templating?.

rsicart commented:
As my first time using this filed, I needed some time before finding the right information.

The link I put is general documentation provided by ansible, like a lot of those filters, which are not provided exclusively by jinja.
For example `regex_search`.

mattclay commented:
/azp run

bmillemathias commented on lib/ansible/plugins/doc_fragments/constructed.py:
```diff
@@ -22,7 +22,10 @@ class ModuleDocFragment(object):
+    description: >
+      Add hosts to group based on Jinja2 conditionals.
+      A list of filters can be found here
+      U(https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html).
```
Instead of using an external link which points to `latest` version so might not be accurate from another version please use an internal link https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role

acozine replied:
Documentation of the correct formats for internal links: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation.

rsicart replied:
I fixed as requested, please let me know if that's what you expect.
Thanks !

acozine reviewed (commented):
This link seems useful, thanks for the idea @rsicart. Can you fix the format to use an internal link (which will keep the link alive if we move that content in future)?

Commit:
Fix: use internal link

As described in:

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html

samccann commented:
/azp run

samccann commented:
Hi @rsicart ! Sorry for the delays and changes here. The updated requirements for crossreferences require that you use R() as specified in https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block

Can you update the PR for that? I'm guessing it needs a rebase by now as well. Again, sorry for all the delays here and shifting requirements. The move to collections complicated how we add crossreferences to  module docs.
